libtransmission/verify.cc

```diff
@@ -45,6 +45,8 @@ void tr_verify_worker::verify_torrent(Mediator& verify_mediator, std::atomic<boo
     auto buffer = std::vector<std::byte>(1024U * 256U);
     auto sha = tr_sha1::create();
     auto last_slept_at = current_time_secs();
+    bool file_valid = false;
+    std::vector<tr_file_index_t> files_ending_in_current_piece;
```

Comment:
It feels like the piece-to-file housekeeping here and in verify_callback() are repeating some code that's already written + unit tested in `file-piece-map.h` but is currently hidden behind a `#error only libtransmission should #include this header.` compile guard. 

@srussel WWYT if `tr_torrent_metainfo` had the `file_piece_map::file_span()` function so that you could get a range of files associated with a piece? Would we even still need `files_ending_in_current_piece` or `on_file_ok()`?

Reply:
@srussel ping

Reply:
I do intend to finish this but I won't have time to work on it until mid-December.

Reply:
> @srussel what would you think if `tr_torrent_metainfo` had the `file_piece_map::file_span()` function so that you could get a range of files associated with a piece? Would we even still need `files_ending_in_current_piece` or `on_file_ok()`?

I have not looked into this yet. I wanted to push an update with all the other issues addressed first before I do any significant changes to the verify function so I have something working that I can compare it to.

Reply:
> I have not looked into this yet. I wanted to push an update with all the other issues addressed first before I do any significant changes to the verify function so I have something working that I can compare it to.

@srussel Is this still on-going?

Reply:
No, I've been busy again with other things. Is this something you need me to do in order to merge this PR? It's just that it could be a while before I get the time to work on it.